cli/azd/pkg/config/manager.go

```diff
@@ -114,3 +115,42 @@ func GetUserConfigFilePath() (string, error) {
 
 	return filepath.Join(configPath, "config.json"), nil
 }
+
+// GetUserConfig gets the Config for the current user, given a manager. If the config file does not exist, an empty
+// configuration object is returned.
+func GetUserConfig(configManager Manager) (Config, error) {
```

Comment:
Having static helpers like these feels a little backwards to me, and we should generally avoid them.

The way I would love to see this structured:

- We have a general purpose `Configuration` class that supports saving and loading config settings. Ideally, I think the filepath should be moved to `Configuration` object construction, because it feels like if `Save` and `Load` allowed different filepaths, it reads more like a filesystem abstraction than a configuration file abstraction. The `Configuration` class would have a constructor that accepts the filepath, and now the API for using that `Configuration` is simplified to `Save` and `Load` without filepath being arguments.
- Now, we probably want a single config file for `azd`. So, we inject a single `*ConfigManager` class: `UserConfigManger` that is bounded to the single user-config filepath.

Reply:
FWIW - I also like this design more - I thought about a similarish refactoring but ended up not doing it here because of the scope creep.  I'll do some fixes here.